open-banking-accelerator/components/consent-management/com.wso2.openbanking.accelerator.consent.extensions/src/main/java/com/wso2/openbanking/accelerator/consent/extensions/authorize/impl/DefaultConsentRetrievalStep.java

```diff
@@ -101,30 +98,28 @@ public void execute(ConsentData consentData, JSONObject jsonObject) throws Conse
             JSONArray accountsJSON = ConsentRetrievalUtil.appendDummyAccountID();
             jsonObject.appendField(ConsentExtensionConstants.ACCOUNTS, accountsJSON);
 
-
         } catch (ConsentException e) {
             JSONObject errorObj = (JSONObject) e.getPayload();
             JSONArray errorList = (JSONArray) errorObj.get("Errors");
             jsonObject.put(ConsentExtensionConstants.IS_ERROR,
                     ((JSONObject) errorList.get(0)).getAsString("Message"));
             return;
-        } catch (ConsentManagementException e) {
+        } catch (ParseException | ConsentManagementException e) {
             throw new ConsentException(ResponseStatus.INTERNAL_SERVER_ERROR,
                     "Exception occurred while getting consent data");
         }
     }
+
     /**
      * Method to retrieve consent related data from the initiation payload.
      * @param consentResource
-     * @return
+     * @return  consent
      * @throws ConsentException
      */
     public JSONArray getConsentDataSet(ConsentResource consentResource)
-            throws ConsentException {
+            throws ConsentException, ConsentManagementException, ParseException {
 
-        String type = consentResource.getConsentType();
-        consentRetrievalHandler = AcceleratorConsentExtensionFactory.getConsentRetrievalHandler(type);
-        return consentRetrievalHandler.getConsentDataSet(consentResource);
```

Comment:
Since we are discussed to remove `ConsentRetrievalHandler` , shall we remove it from this same PR?

Reply:
Fixed in [fa29ae7]